runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/DataflowPipelineTranslatorTest.java

```diff
@@ -676,119 +654,33 @@ public void process(ProcessContext c) {
-    DataflowPipelineOptions options = buildPipelineOptions();
-    DataflowRunner runner = DataflowRunner.fromOptions(options);
-    options.setStreaming(true);
-    options.setExperiments(Arrays.asList("beam_fn_api"));
-    DataflowPipelineTranslator translator = DataflowPipelineTranslator.fromOptions(options);
-
-    Pipeline pipeline = Pipeline.create(options);
-
-    PCollection<String> windowedInput =
-        pipeline
-            .apply(Create.of("a"))
-            .apply(Window.into(FixedWindows.of(Duration.standardMinutes(1))));
-    windowedInput.apply(ParDo.of(new TestSplittableFn()));
-
-    runner.replaceTransforms(pipeline);
-
-    JobSpecification result = translator.translate(pipeline, runner, Collections.emptyList());
-
-    Job job = result.getJob();
-
-    // The job should contain a SplittableParDo.ProcessKeyedElements step, translated as
-    // "SplittableProcessKeyed".
+    // The job should look like:
+    // 0. ParallelRead (Create)
+    // 1. ParallelDo
 
     List<Step> steps = job.getSteps();
-    Step processKeyedStep = null;
-    for (Step step : steps) {
-      if ("SplittableProcessKeyed".equals(step.getKind())) {
-        assertNull(processKeyedStep);
-        processKeyedStep = step;
-      }
-    }
-    assertNotNull(processKeyedStep);
+    assertEquals(2, steps.size());
 
-    String fn = Structs.getString(processKeyedStep.getProperties(), PropertyNames.SERIALIZED_FN);
+    Step createStep = steps.get(0);
+    assertEquals("ParallelRead", createStep.getKind());
 
-    Components componentsProto = result.getPipelineProto().getComponents();
-    RehydratedComponents components = RehydratedComponents.forComponents(componentsProto);
-    RunnerApi.PTransform spkTransform = componentsProto.getTransformsOrThrow(fn);
-    assertEquals(
-        PTransformTranslation.SPLITTABLE_PROCESS_KEYED_URN, spkTransform.getSpec().getUrn());
-    ParDoPayload payload = ParDoPayload.parseFrom(spkTransform.getSpec().getPayload());
-    assertThat(
-        ParDoTranslation.doFnAndMainOutputTagFromProto(payload.getDoFn()).getDoFn(),
-        instanceOf(TestSplittableFn.class));
-    assertThat(
-        components.getCoder(payload.getRestrictionCoderId()), instanceOf(SerializableCoder.class));
-
-    // In the Fn API case, we still translate the restriction coder into the RESTRICTION_CODER
-    // property as a CloudObject, and it gets passed through the Dataflow backend, but in the end
-    // the Dataflow worker will end up fetching it from the SPK transform payload instead.
-    Coder<?> restrictionCoder =
-        CloudObjects.coderFromCloudObject(
-            (CloudObject)
-                Structs.getObject(
-                    processKeyedStep.getProperties(), PropertyNames.RESTRICTION_CODER));
-    assertEquals(SerializableCoder.of(OffsetRange.class), restrictionCoder);
+    Step parDoStep = steps.get(1);
+    assertEquals("ParallelDo", parDoStep.getKind());
```

Comment:
You should check that this ParallelDo has the restriction coder on it otherwise Dataflow will not know how to perform the expansion.

Reply:
Would this be part of the `Step.properties` map? Inspecting under a debugger I currently see properties:

* `serialized_fn`
* `non_parallel_inputs`
* `user_name`
* `display_data`
* `parallel_input`
* `output_info`
    * `output_name`
    * `user_name`
    * `encoding`
        * `@type`
        * `is_wrapper`
        * `component_encodings`

I suspect the restriction coder is something I still need to add.

Reply:
Oh, actually it looks like I'll need to crack open the serialized dofn and inspect the embedded restriction coder. The [previous test case](https://github.com/apache/beam/pull/7001/files/9928e6d5e6b6516a7609e27e5e9fffb7794055c5#diff-9d2a89b05baa928a7468701ec3afa176L712) shows how.